lib/inspec/resources/auditd.rb

```diff
@@ -187,8 +187,20 @@ def syscalls_for(line)
       line.scan(/-S ([^ ]+)\s?/).flatten.first.split(",")
     end
 
+    # Processes the line and returns a pair of entries reflecting the 'action'
+    # and 'list' items.
+    #
+    # @return [Array[String,String]]
     def action_list_for(line)
-      line.scan(/-a ([^,]+),([^ ]+)\s?/).flatten
+      action_list = line.scan(/-a ([^,]+),([^ ]+)\s?/).flatten
+
+      # Actions and lists can be in either order
+      valid_actions = %w{never always}
+
+      [
+        (action_list & valid_actions).first,
+        (action_list - valid_actions).first
+      ]
     end
```

Comment:
> Ok, so I looked at the testing structure but the functions aren't unit tested directly and I couldn't figure out how to do red/green testing with the existing code.

You can just write very plain unit tests for this that pass in the line and assert what you expect on the way out. Something like:

```ruby
resource = MockLoader.new(:centos7).load_resource("auditd") # which I don't like but it's there...
assert_equal ["a", "b"], resource.action_list_for("input line")
```

As long as you have covered both the positive and negative edge cases, then I think this is fine.